algo/src/main/java/com/hartwig/serve/ServeAlgo.java

```diff
@@ -173,13 +184,35 @@ private ExtractionResult extractHartwigCohortKnowledge(@NotNull String hartwigCo
     private ExtractionResult extractHartwigCuratedKnowledge(@NotNull String hartwigCuratedTsv, boolean addExplicitHotspots)
             throws IOException {
         LOGGER.info("Reading Hartwig Curated TSV from '{}'", hartwigCuratedTsv);
-        List<HartwigEntry> entries = HartwigFileReader.read(hartwigCuratedTsv);
+        List<HartwigHotspotEntry> entries = HartwigHotspotFileReader.read(hartwigCuratedTsv);
         LOGGER.info(" Read {} entries", entries.size());
 
         RefGenomeResource refGenomeResource = refGenomeManager.pickResourceForKnowledgebase(Knowledgebase.HARTWIG_CURATED);
-        HartwigExtractor extractor =
-                new HartwigExtractor(Knowledgebase.HARTWIG_CURATED, refGenomeResource.proteinResolver(), addExplicitHotspots);
+        HartwigHotspotExtractor extractor =
+                new HartwigHotspotExtractor(Knowledgebase.HARTWIG_CURATED, refGenomeResource.proteinResolver(), addExplicitHotspots);
         LOGGER.info("Running Hartwig Curated knowledge extraction");
         return extractor.extract(entries);
     }
+
+    @NotNull
+    private ExtractionResult extractHartwigDriverGeneKnowledge(@NotNull String driverGeneFileTsv) throws IOException {
+        LOGGER.info("Reading Hartwig Driver Genes TSV from '{}'", driverGeneFileTsv);
+        List<HartwigGeneEntry> entries = HartwigGeneFileReader.read(driverGeneFileTsv);
```

Comment:
we could also read directly from the `DriverGeneFile` here and just pluck out the gene. Perhaps that's more robust, but felt inconsistent. What do you think @kduyvesteyn ?

Reply:
I prefer your solution. There is a somewhat sneaky dependency that both input files have a column called 'gene' but I think that's a fairly robust assumption anyways (also considering both files are managed by HMF).